### DISCUSSION TITLE
Possible improvements for user executors

### DISCUSSION THREAD

**Discussion by manzan46** [ovh/venom#369] [Ideas]
👋  Hope this is the right place for improvement ideas :) 

We recently at @heetch try to find a tool to replace our current homemade testing tool and had a bit of test with Venom 🙌 

The recent addition of `User executors` definitely makes it easier with reusability. 

Few things that could help even more: 

1️⃣  First would be able to pass to the command a flag to specify the `lib` folder directory, if nothing is specified we keep the current behavior of `lib`

This would allow more flexibility, in order to share user_executor if you have testsuits in different folders (e.g. microservices)

2️⃣  Second one is to make `getFilePaths` using `filepath.Walk` to get all yaml file in the directory and subdirectory

This would allow to not have all user executors in one folder. 

3️⃣  When a user executor assertions failed, in the output, we just get a `FAILURE` message but not the assertion failure as we have for teststep, so we have to dig in the log.

--- 

Hope this makes sense, I got 1️⃣  and 2️⃣  working on my local build so happy to open PR if you think this would help, for 3️⃣  I got something kind of working but it's more of a hack, and the display is not great 🙈   

![image](https://user-images.githubusercontent.com/1071962/109401927-ff737280-7951-11eb-96aa-06a7cb341f14.png)

**Comment by fsamin** [comment]
Hello, 
Thanks for you feedback. 
Those suggestions sound good. Can you create 3 new issues to track them ?

**Reply by manzan46** [reply]
Thanks will do 🙏🏻

**Reply by yesnault** [reply]
1️⃣ 2️⃣ 3️⃣ now on master

**Reply by manzan46** [reply]
🔥  🥳  🙏🏻  `:parrot-party:`
